File: problems/1123/solution.go
package main
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func lcaDeepestLeaves(root *TreeNode) *TreeNode {
    node, _ := find(root)
    
    return node
}

func find(node *TreeNode) (*TreeNode, int) {
    if node == nil {
        return node, 0
    }
    
    lnode, lh := find(node.Left)
    rnode, rh := find(node.Right)
    
    if lh == rh {
        return node, lh + 1
    } else if lh > rh {
        return lnode, lh + 1
    }
    
    return rnode, rh + 1
}

File: problems/1123/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def lcaDeepestLeaves(self, root: TreeNode) -> TreeNode:
        node, _ = self.find(root)
        
        return node
    
    def find(self, node):
        if not node:
            return None, 0
        
        lnode, lh = self.find(node.left)
        rnode, rh = self.find(node.right)
        
        if lh == rh:
            return node, lh + 1
        elif lh > rh:
            return lnode, lh + 1
        
        return rnode, rh + 1